WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController+JetpackInstall.swift

```diff
@@ -2,34 +2,66 @@ import UIKit
 
 extension MySiteViewController {
     func subscribeToJetpackInstallNotifications() {
-        NotificationCenter.default.addObserver(self, selector: #selector(launchJetpackInstallFromNotification(_:)), name: .installJetpack, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(promptJetpackPluginInstallFromNotification(_:)), name: .promptInstallJetpack, object: nil)
     }
 
-    /// Presents the Jetpack Login (Install) controller from a notification
-    @objc func launchJetpackInstallFromNotification(_ notification: NSNotification) {
-        guard let blog = blog else {
+    /// Presents the prompt to install Jetpack from a notification
+    @objc func promptJetpackPluginInstallFromNotification(_ notification: NSNotification) {
+        guard let blog = blog,
+            let promptSettings = notification.userInfo?["jetpackInstallPromptSettings"] as? JetpackInstallPromptSettings else {
             return
         }
 
-        let controller = JetpackLoginViewController(blog: blog)
-        controller.promptType = .installPrompt
+        let installPromptViewController = JetpackInstallPromptViewController(blog: blog)
+        let installPromptNavigationController = UINavigationController(rootViewController: installPromptViewController)
+        installPromptNavigationController.modalPresentationStyle = .fullScreen
 
-        let navController = UINavigationController(rootViewController: controller)
-        navController.modalPresentationStyle = .fullScreen
+        installPromptViewController.actionHandler = { [weak self] action in
+            switch action {
+            case .noThanks:
+                installPromptNavigationController.dismiss(animated: true)
+            case .install:
+                self?.presentJetpackInstall(on: installPromptNavigationController, blog: blog, jetpackInstallPromptSettings: promptSettings)
```

Comment:
Similar to the question above about the need for a navigation controller, should we push this installation flow screen on to`installPromptNavigationController` instead of presenting it on top of it? 

I could see a reason to keep it this way if we want users who cancel some part of the installation flow to return to the prompt. However, as a user, I think if I canceled the installation flow at any point, I'd also want to dismiss the prompt. Wdyt?

Reply:
This is a valid question. 

If we push the installation flow, I thought then it would make sense to allow users to come `< Back` rather than dismiss. Although not allowing them to come back would allow avoiding these additional "partial installation" checks inside the prompt. I'll think about it.